gatling-http/src/main/scala/io/gatling/http/action/ws/WsFrameCheckSequenceBuilder.scala

```diff
@@ -25,13 +25,29 @@ import io.gatling.http.check.ws.{ WsFrameCheck, WsFrameCheckSequence }
 object WsFrameCheckSequenceBuilder {
 
   @SuppressWarnings(Array("org.wartremover.warts.ListAppend"))
-  def resolve[T <: WsFrameCheck](builders: List[WsFrameCheckSequenceBuilder[T]], session: Session): Validation[List[WsFrameCheckSequence[T]]] =
+  def resolve[T <: WsFrameCheck](builders: List[WsFrameCheckSequenceBuilder[T]], session: Session): Validation[List[WsFrameCheckSequence[T]]] = {
     builders.foldLeft(List.empty[WsFrameCheckSequence[T]].success) { (acc, builder) =>
       for {
-        accValue <- acc
         timeout <- builder.timeout(session)
-      } yield accValue :+ WsFrameCheckSequence(timeout, builder.checks)
+        accValue <- acc
+      } yield accValue :+ WsFrameCheckSequence(timeout, resolveCheckName(builder.checks, session).asInstanceOf[List[T]])
     }
+  }
+
+  @SuppressWarnings(Array("org.wartremover.warts.ListAppend"))
+  private def resolveCheckName[T <: WsFrameCheck](checks: List[WsFrameCheck], session: Session): List[WsFrameCheck] = {
```

Comment:
First, I don't like it. I'd rather have 2 sets of types: one for the DSL with the Expression and one for runtime with the resolved value.

Then, you don't properly handle failures. What you must do is a recursive resolution. See here for an example https://github.com/gatling/gatling/blob/15664aa2fbfbf4ac55190eb83115e219a777d529/gatling-core/src/main/scala/io/gatling/core/session/package.scala#L58-L79